fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -252,18 +262,45 @@ private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField>
         }
     }
 
-    private void deleteDocument(Tuple groupingKey, byte[] primaryKey) throws IOException {
+    private byte[][] retrievePoints(@Nonnull Tuple primaryKey) throws IOException {
+        byte[][] lookup = new byte[primaryKey.size()][];
+        for (int i = 0; i < lookup.length; i++) {
+            Object value = primaryKey.get(i);
+            if (value == null) {
+                throw new UnsupportedEncodingException("Null Primary Keys Not Supported in Lucene");
+            }
+            if (value instanceof Long) {
+                lookup[i] = Longs.toByteArray(((Long)value));
+            } else if (value instanceof Integer) {
+                lookup[i] = Ints.toByteArray(((Integer)value));
+            } else if (value instanceof Double) {
+                lookup[i] = ByteBuffer.allocate(8).putDouble((Double) value).array();
+            } else if (value instanceof Boolean) {
+                lookup[i] = ByteBuffer.allocate(1).put( ((Boolean)value) ? (byte)1 : (byte)0).array();
+            } else {
+                throw new UnsupportedEncodingException("Primary Key of Type X Not Supported in Lucene");
+            }
+        }
+        return lookup;
+    }
+
+    private <M extends Message> void deleteDocument(Tuple groupingKey, @Nonnull FDBIndexableRecord<M> oldRecord) throws IOException {
+        Tuple primaryKey = oldRecord.getPrimaryKey();
         final IndexWriter oldWriter = getOrCreateIndexWriter(state, indexAnalyzer, executor, groupingKey);
-        Query query = SortedDocValuesField.newSlowExactQuery(PRIMARY_KEY_SEARCH_NAME, new BytesRef(primaryKey));
```

Comment:
I'm not sure why this is using `newSlowExactQuery`, but should these use `IndexWriter.deleteDocument(Term)` instead with a `Term` containing the primary key field as the field name and the primary key as the value? I don't know how that compares in performance to the binary point approach, but it would seem to be a simpler way of deleting, and it would work for all primary keys

Reply:
Term search is pretty slow if you are trying to delete 50 records for example.